src/simulation.jl

```diff
@@ -253,6 +254,25 @@ function compile_env(d::SummationDecapode, dec_matrices::Vector{Symbol}, con_dec
     push!(defined_ops, op)
   end
 
+  # These are nonoptimizable default DEC functions.
+  for op in nonoptimizable_operators
+    op in defined_ops && continue
+
+    quote_op = QuoteNode(op)
+
+    # TODO: Add support for user-defined code targets
+    default_generation = @match code_target begin
+      ::CPUBackend => :default_dec_generate
+      ::CUDABackend => :default_dec_cu_generate
+      _ => throw(InvalidCodeTargetException(code_target))
+    end
```

Comment:
the API of a code target should be a function `generator_function(::CodeTarget)::Function`

```julia
generator_function(::AbstractGenerationTarget) = default_dec_generate
generator_function(::CUDABackend) = default_dec_cu_generate # not sure if this should be the function or the symbol? you will get the right namespace by using function instead of symbol.
```

then at this location you do 

```julia
default_generation = $(generator_function(code_target))
```
PS: this match statement is currently inside a loop, but could be hoisted out of the loop, because it doesn't depend on the loop variable.